edifact/models/tests/base_segment_test.py

```diff
@@ -1,40 +1,64 @@
 import abc
-
 from edifact.edifact_exception import EdifactValidationException
-from edifact.outgoing.models.segment import Segment
+from edifact.models.segment import Segment
 
 
 class BaseSegmentTest(abc.ABC):
     """
     A base test for subclasses of Segment. Implementor must also inherit unittest.TestCase.
     """
-
     @abc.abstractmethod
     def _create_segment(self) -> Segment:
         """
         :return: a valid Segment with all attributes populated
+        :return: a valid Segment with all attributes populated, created with given parameters
         """
         pass
 
+    @abc.abstractmethod
+    def _create_segment_from_string(self) -> Segment:
+        """
+        :return: a valid Segment with all attributes populated, generated from edifact segment string
+        """
+
     @abc.abstractmethod
     def _get_attributes(self):
         """
         :return: all of the Segments settable attributes
         """
         pass
-    
+
+    def _get_expected_segment(self):
+        """
+        :return: the expected segment from EDIFACT returned by _create_segment_from_string()
+        """
+        return self._create_segment()
+
     @abc.abstractmethod
     def _get_expected_edifact(self):
         """
         :return: the expected EDIFACT for the segment returned by _create_segment()
         """
         pass
 
+    @abc.abstractmethod
+    def _compare_segments(self, expected_segment: Segment, actual_segment: Segment) -> bool:
+        """
+        compares two segment's attributes one by one
```

Comment:
It doesn't actually do this. It compares the two segments and raises an (AssertionError? I'm not sure of the actual type used by unittest) if they are not the same